### PR TITLE
Add repr for plugins

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
None

### Summary of the issue:
Several plugin types, such as addon objects and drivers, have a default repr that makes it impossible to identify the addon from the python console without further inspection.

### Description of user facing changes
When inspecting addons, drivers and appModules from the python console, the repr will be different.

### Description of development approach
Added `__repr__` method to several classes

### Testing strategy:
Tested inspection of the objects using the python console.

### Known issues with pull request:
None known
### Change log entries:
For Developers
- Some plugin objects (e.g. drivers and add-ons) now have a more informative description in the python console.

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

File: source/appModuleHandler.py [hunk 1]
```diff
@@ -563,7 +563,10 @@ def _get_productVersion(self):
 		return self.productVersion
 
 	def __repr__(self):
-		return "<%r (appName %r, process ID %s) at address %x>"%(self.appModuleName,self.appName,self.processID,id(self))
```
Comment (seanbudd) [inline source/appModuleHandler.py:566]: is removing the address information intended?

**Review by seanbudd — APPROVED** [review]
Generally LGTM

File: source/appModuleHandler.py (hunk 1 above)
Comment (leonardder) [reply source/appModuleHandler.py:566]: Yes. I've never seen a case where having the address in repr made any sense. I can easily restore it though.

Comment (seanbudd) [reply source/appModuleHandler.py:566]: makes sense to me

**Pull request merged**